cms-2017-simulated-datasets/code/dataset_records.py

```diff
@@ -0,0 +1,654 @@
+    rec['abstract']['description'] = '<p>' + additional_title + '.</p>' + \
+                                     '<p>See the description of the simulated dataset names in: <a href="/about/CMS-Simulated-Dataset-Names">About CMS simulated dataset names</a>.</p>' + \
+                                     '<p>These simulated datasets correspond to the collision data collected by the CMS experiment in ' + year_created + '.</p>'
+
+    rec['accelerator'] = "CERN-LHC"
+
+    rec['collaboration'] = {}
+    rec['collaboration']['name'] = 'CMS Collaboration'
+    # rec['collaboration']['recid'] = get_author_list_recid(dataset_full_name)
+
+    rec['collections'] = ['CMS-Simulated-Datasets', ]
+
+    rec['collision_information'] = {}
+    rec['collision_information']['energy'] = collision_energy
+    rec['collision_information']['type'] = collision_type
+
+    # if dataset_format == "NANOAODSIM":
+        # dataset_path = f"/eos/opendata/cms/mc/{dataset_runperiod}/{dataset}/NANOAODSIM/{dataset_version}"
+        # intermediate_dir = os.listdir(dataset_path)
+        # sample_file_path = f"{dataset_path}/{intermediate_dir[0]}"
+        # sample_file_with_path = f"{sample_file_path}/{os.listdir(sample_file_path)[0]}"
+        #rec["dataset_semantics_files"] = get_dataset_semantics_doc(dataset, sample_file_with_path, str(recid_info[dataset_full_name]))
+
+    rec['date_created'] = [year_created]
+    rec['date_published'] = year_published
+    rec['date_reprocessed'] = get_data_processing_year(dataset_full_name, mcm_dir)
+    rec['distribution'] = {}
+    rec['distribution']['formats'] = [dataset_format.lower(), 'root']
+    rec['distribution']['number_events'] = get_number_events(dataset_full_name, das_dir)
+    rec['distribution']['number_files'] = get_number_files(dataset_full_name, das_dir)
+    rec['distribution']['size'] = get_size(dataset_full_name, das_dir)
+
+    doi = get_doi(dataset_full_name, doi_info)
+    if doi:
+        rec['doi'] = doi
+
+    rec['experiment'] = ['CMS', ]
+
+    rec_files = get_dataset_index_files(dataset_full_name, eos_dir)
+    if rec_files:
+        rec['files'] = [] 
+        for index_type in ['.json', '.txt']:
+            index_files = [f for f in rec_files if f[0].endswith(index_type)]
+            for file_number, (file_uri, file_size, file_checksum) in enumerate(index_files):
+                rec['files'].append({
+                    'checksum': 'adler32:' + file_checksum,
+                    'description': dataset + dataset_format + ' dataset file index (' + str(file_number + 1) + ' of ' + str(len(index_files)) + ') for access to data via CMS virtual machine',
+
+                    'size': file_size,
+                    'type': 'index' + index_type,
+                    'uri': file_uri
+                })
+
+    rec['license'] = {}
+    rec['license']['attribution'] = 'CC0'
+
+    rec['methodology'] = get_all_generator_text(dataset_full_name, das_dir, mcm_dir, conffiles_dir, recid_info)
```

Comment:
This is the time-consuming part and it is exactly the same for mini and nano.
Check if the related nano dataset record has already been built take the full methodology from there.